### PR TITLE
fix(tmdb): resolve AI/box-set disc titles that carry a season subtitle

### PR DESCRIPTION
**Pull request by Jsakkos**
## Summary

A TV disc whose AI-guessed (or box-set) title includes the **season's subtitle** — e.g. Gemini returning `"Avatar: The Last Airbender Book One: Water"` for label `Avatar_Book_1_Disc_1` — failed every TMDB lookup, because TMDB indexes the series as `"Avatar: The Last Airbender"` (id 246) and the over-specified string returns **0 results**. That left `tmdb_id` null, which cascaded into three user-visible symptoms:

1. an unnecessary identity prompt (Gate B "TV-without-TMDB"),
2. a "correct-looking" name that nonetheless never resolved, and
3. a total subtitle-download failure (`ValueError: Could not find show ... on TMDB`).

Root cause and fix verified against the live TMDB API: the full string → 0 results; `"Avatar: The Last Airbender"` → id 246.

## What changed

- **`generate_name_variations` (`backend/app/matcher/tmdb_client.py`)** — add a variation that strips a trailing `Book/Volume/Vol/Part/Pt/Season/Series/Chapter <count>` subtitle. This one function is shared by `fetch_show_id` (subtitle path), `classify_from_tmdb` (identify-time AI re-query), and the resume-time `_resolve_missing_tmdb_id`, so the fix lands once and covers all three resolution paths. The marker must be followed by a digit / ordinal word / **uppercase** roman numeral, so legitimate titles ("Part of Me", "Doctor Who Part Mix") are never truncated.
- **`IDENTIFICATION_PROMPT` (`backend/app/core/ai_identifier.py`)** — instruct the model to return the **series title only** for TV (no season/book/volume/part subtitle), reducing over-specification at the source.
- Tests + a CHANGELOG `[Unreleased]` entry.

Net diff is +115 lines, fully additive — no production logic removed.

## Why no new tmdb_id search path

Subtitle search already prefers `tmdb_id` (OpenSubtitles `parent_tmdb_id`) and falls back to name. The failure wasn't a missing id-based path — it was that **no id ever resolved**. Once the variation resolves id 246, the existing tmdb_id path works.

## Test plan

- [x] `uv run pytest tests/unit/test_tmdb_client.py tests/unit/test_tmdb_classifier.py tests/unit/test_ai_identifier.py` → 92 passed
- [x] `uv run ruff check` clean on touched files
- New regression tests: subtitle stripping (Book/Volume/Part), guard against truncating "Part of Me" and the lowercase-roman-letter word case ("Part Mix"), and `classify_from_tmdb` recovery via the stripped variation (exercises the real fallback loop, not a mocked `classify_from_tmdb`).

## Reviewer notes

- The new rule only *adds* a search variation tried **after** the exact name fails, so blast radius is low even in the unlikely false-positive case.
- Full design + task breakdown: `docs/superpowers/plans/2026-06-17-ai-name-tmdb-resolution.md` (included in this PR).
- Not included here: a one-off data rescue for the in-flight job (`Avatar_Book_1_Disc_1`, tmdb_id null) on the user's running v0.21.3 build — that's re-identifying the live job, separate from this code change.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: backend/app/matcher/tmdb_client.py [hunk 1]
```diff
@@ -221,6 +240,11 @@ def generate_name_variations(name: str) -> list[str]:
         if before_dash and before_dash != current:
             variations.append(before_dash)
 
+    # Remove a trailing season/box-set subtitle (see _TRAILING_SET_SUBTITLE_RE).
+    set_subtitle_cleaned = _TRAILING_SET_SUBTITLE_RE.sub("", current).strip()
+    if set_subtitle_cleaned and set_subtitle_cleaned != current:
+        variations.append(set_subtitle_cleaned)
```
[the remark this answers is not in the document]
Comment (Jsakkos) [reply backend/app/matcher/tmdb_client.py]: Fixed in 7a5a65d1. The set-subtitle block now trims a dangling separator after stripping (`.rstrip(" -:")`), so `"Show - Part Two"` yields `"Show"` instead of the junk `"Show -"`. Note: applying the regex to `before_dash` alone would not have removed the junk, since `before_dash` is already `"Show"` — the orphaned `"Show -"` came from the `current`-based strip. Added `test_dash_joined_subtitle_yields_clean_name`.

File: backend/tests/unit/test_tmdb_classifier.py [hunk 2]
```diff
@@ -244,3 +245,33 @@ def test_single_char_tokens_filtered(self):
...
+
+    tv_hit = {
+        "id": 246,
+        "name": "Avatar: The Last Airbender",
+        "first_air_date": "2005-02-21",
+        "popularity": 100.0,
+        "original_name": "Avatar: The Last Airbender",
+    }
+
+    def fake_search(url, name, headers, base_params, timeout):
```
[the remark this answers is not in the document]
Comment (Jsakkos) [reply backend/tests/unit/test_tmdb_classifier.py]: Fixed in 7a5a65d1 — renamed the `fake_search` parameter `name` → `query` to match the real `_search_tmdb` signature.

File: backend/tests/unit/test_tmdb_client.py [hunk 3]
```diff
@@ -762,3 +763,48 @@ def test_path_injecting_group_id_rejected_without_request(self, mock_get):
...
+        # Under IGNORECASE the roman-numeral arm must NOT match a lowercase
+        # word built only from roman-numeral letters ("mix" = m,i,x). A title
+        # like "Doctor Who Part Mix" must not be truncated to "Doctor Who".
+        variations = generate_name_variations("Doctor Who Part Mix")
+        assert "Doctor Who" not in variations
+
+    def test_still_strips_uppercase_roman_numeral(self):
+        # Genuine uppercase roman-numeral season markers still strip.
+        variations = generate_name_variations("Fargo Part II")
+        assert "Fargo" in variations
```
[the remark this answers is not in the document]
Comment (Jsakkos) [reply backend/tests/unit/test_tmdb_client.py:810]: Fixed in 7a5a65d1 — added regression tests for the `Vol.`, `Pt.`, and `Series` arms and the dash edge case, and extended recognized ordinal words through Twenty (`Doctor Who Series Thirteen` now strips; numeric form was already covered by `\d+`).
